jstz_node/src/rollup.rs

```diff
@@ -58,4 +60,43 @@ impl RollupClient {
             Err(anyhow!("Unhandled response status: {}", res.status()))
         }
     }
+
+    pub async fn get_subkeys(&self, key: &str) -> Result<Option<Vec<String>>> {
+        let res = self
+            .client
+            .get(format!(
+                "{}/global/block/head/durable/wasm_2_0_0/subkeys?key={}",
+                self.endpoint, key
+            ))
+            .send()
+            .await?;
+
+        if res.status() == 200 || res.status() == 500 {
+            let content_str = res.text().await?;
+            let content_json = serde_json::from_str::<Value>(&content_str);
+
+            match content_json {
+                Ok(serde_json::Value::Array(arr)) => {
+                    let list_of_strings: Result<Vec<String>> = arr
+                        .into_iter()
+                        .map(|item| match item {
+                            Value::String(s) => Ok(s),
+                            _ => Err(anyhow!("Non-string element found in the array")),
+                        })
+                        .collect();
+
+                    match list_of_strings {
+                        Ok(list) => Ok(Some(list)),
+                        Err(e) => Err(e),
+                    }
+                }
+                Ok(_) => Err(anyhow!(
+                    "Expected a JSON array but got a different structure"
+                )),
+                Err(e) => Err(anyhow!("Failed to parse content as JSON: {:?}", e)),
+            }
```

Comment:
You're manually deserializing here. It would be best to write a rust type (e.g. `SubkeysResponse`) that derives a serializer and deserializer for this type (like `ValueResponse`)